Utilities/FileIO.hpp

```diff
@@ -251,13 +252,24 @@ void readMatPowerBaseMVA(SystemModelData<RealT, IdxT>& mp, std::string& line)
 
 namespace GridKit
 {
+
+/**
+ * @brief Reads a file in for tabulated data
+ *
+ * @todo needs to return int for file error codes
+ * 
+ * @tparam ScalarT 
+ * @param table 
+ * @param filename 
+ * @param ti 
+ * @param tf 
+ */
 template <typename ScalarT>
 void setLookupTable(std::vector<std::vector<ScalarT>>& table,
-                    std::string filename,
+                    std::istream& idata,
                     ScalarT& ti,
                     ScalarT& tf)
```

Comment:
Are these references necessary to ti and tf? The tables format provides them in the first and last locations.

Reply:
I agree it may look redundant but I think these times should be part of the function signature. That way user does not need to know about the structure of the input file and/or how initial/final times are evaluated (e.g., at some point we might want to interpolate time series in the middle of data range).